src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -292,6 +298,14 @@ func (r *Cluster) validateKafkaListeners() field.ErrorList {
 			p.TLS.NodeSecretRef,
 			field.NewPath("spec").Child("configuration").Child("kafkaApi").Index(i).Child("tls"))
 		allErrs = append(allErrs, tlsErrs...)
+
+		// TODO(#2256): Add support for external listener + TLS certs for IPs
```

Comment:
I think it's doable and could be added. I selected not to support it in this PR because there isn't immediate demand and preferred to keep the PR simpler.